jbmc/unit/java_bytecode/java_types/java_method_type.cpp

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************\
+
+ Module: Unit tests for java_types
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <java_bytecode/java_types.h>
+
+SCENARIO("java_method_type", "[core][java_types]")
+{
+  GIVEN("A code_typet")
+  {
+    code_typet::parameterst parameters;
+    typet return_type;
+    code_typet code_type = code_typet();
+    THEN("It has id code_typet and not java_method_typet")
+    {
+      REQUIRE(code_type.id() == ID_code);
+      REQUIRE_FALSE(code_type.get_bool(ID_C_java_method_type));
+    }
+
+    WHEN("It is converted to a java_method_typet")
+    {
+      THEN("It should have id code_typet and java_method_typet")
+      {
+        java_method_typet method_type = to_java_method_type(code_type);
```

Comment:
This test will fail when you introduce the change to not change this on `to_method_type`. I don't think you need this test at all. A more useful test might be in one of the java_bytecode unit tests to check that you get a `java_method_typet` for the method symbols.